04_numeronym/davidbroughsmyth/main.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
+
+func numeronyms(vals ...string) []string {
+	result := make([]string, len(vals))
+	for i, val := range vals {
+		if len(val) > 3 {
+			result[i] = val[0:1] + strconv.Itoa(len(val)-2) + val[len(val)-1:]
+		} else {
```

Comment:
Rather than an `else` here, you could just assign to `val` in the `if` case and assign to `result[i]` unconditionally after it. That more clearly expresses the intent that the value is modified if the length is greater than 3, and is unchanged otherwise. The assignment to `result[i]` is unconditional. Just a small point but a lot of Go code tries to keep as much code further left as possible, leaving the indented code as the more special cases.

Reply:
Done.